boa/util/deploy_cache.py

```diff
@@ -0,0 +1,80 @@
+import json
+from functools import cached_property
+from hashlib import sha256
+from pathlib import Path
+from sqlite3 import connect
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from boa.network import TraceObject
+
+
+class DeployCache:
+    """
+    A cache for storing deploy information.
+    """
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    @cached_property
+    def connection(self):
+        self.path.parent.mkdir(parents=True, exist_ok=True)
+        connection = connect(self.path)
+        # return results as dictionaries
+        connection.row_factory = lambda cursor, row: {
+            column[0]: row[index] for index, column in enumerate(cursor.description)
+        }
+        connection.execute(
+            """
+            CREATE TABLE IF NOT EXISTS deploy_cache (
+                deploy_id TEXT, hash TEXT, -- cache key
+                receipt TEXT, trace TEXT, -- cache value
+                PRIMARY KEY(deploy_id, hash)
+            )
+        """
+        )
+        return connection
+
+    def get(
+        self, source_code: str, bytecode: bytes, deploy_id: str, chain_id: int
+    ) -> tuple[dict, "TraceObject"] | tuple[None, None]:
+        if not deploy_id or not source_code:
+            return None, None
+
+        hash_str = self._hash_str(bytecode, source_code, chain_id)
```

Comment:
hmm, i don't think this is going to work

Reply:
I'm pretty sure it works, but maybe you were expecting something different. Please specify what you guys went to be implemented in #82